Lib/ipaddress.py

```diff
@@ -1834,8 +1854,23 @@ def __init__(self, address):
         addr_str = str(address)
         if '/' in addr_str:
             raise AddressValueError("Unexpected '/' in %r" % address)
+        addr_str, self._scope_id = self._split_scope_id(addr_str)
+
         self._ip = self._ip_int_from_string(addr_str)
 
+    def __str__(self):
+        ip_str = super().__str__()
+        return ip_str + '%' + self._scope_id if self._scope_id else ip_str
+
+    @property
+    def scope_id(self):
+        """Identifier of a particular zone of the address's scope.
+
+        Returns:
+            A string, identifying the zone of the address if specified, else None.
+        """
```

Comment:
Let's mention RFC 4007 in the docstring. This concept will be unfamiliar to many people (even people who know a bit about IPv6), so a reference will be helpful.

Reply:
[Fixed](https://github.com/python/cpython/pull/13772/commits/b0a4fe901b2300b0be57619bf30d60e904e43fec)